compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/util/NumericLiteralSupport.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * you may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.ballerinalang.compiler.util;
+
+import java.math.BigDecimal;
+import java.math.MathContext;
+
+/**
+ * Common functions used for handling discriminated float/decimal literals.
+ */
+public class NumericLiteralSupport {
```

Comment:
- [ ] Better to make this final class, as it is not intended to extend. 
- [ ] Add java docs to all public methods.